app/javascript/components/Profile/DeleteAccount.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { InputOnChangeData, Message, Segment, Form, Header } from 'semantic-ui-react';
+import { inject, observer } from 'mobx-react';
+import SessionStore, { RequestState } from '../../stores/session_store';
+
+
+interface InjectedProps {
+  sessionStore: SessionStore;
+}
+
+@inject('sessionStore')
+@observer
+export default class DeleteAccount extends React.Component {
+  private password: string = '';
+
+  get injected() {
+    return this.props as InjectedProps;
+  }
+
+  onChangePassword = (event: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => {
+    this.password = event.target.value;
+  }
+
+  deleteAccount() {
+    this.injected.sessionStore.deleteAccount(this.password);
+  }
+
+  render() {
+    const { passwordState } = this.injected.sessionStore;
+    return (
+      <Segment
+        color="red"
+        style={{ minWidth: '350px' }}
+      >
+        <Header
+          as="h2"
+          content="Parmanently delete your Account."
+        />
+        <Message
+          info
+          icon="info"
+          content="All your data is deleted permanently and can not be restored."
```

Comment:
I would rather use **cannot** here: https://en.oxforddictionaries.com/usage/cannot-or-can-not